linkcheck/htmlutil/linkparse.py

```diff
@@ -92,44 +92,6 @@ def strip_c_comments (text):
     return c_comment_re.sub('', text)
 
 
-class StopParse(Exception):
-    """Raised when parsing should stop."""
-    pass
-
-
-class TagFinder (object):
-    """Base class handling HTML start elements.
-    TagFinder instances are used as HTML parser handlers."""
-
-    def __init__ (self):
-        """Initialize local variables."""
-        super(TagFinder, self).__init__()
-
-    def start_element (self, tag, attrs, element_text, lineno, column):
-        """Does nothing, override in a subclass."""
-        pass
-
-
-class MetaRobotsFinder (TagFinder):
-    """Class for finding robots.txt meta values in HTML."""
-
-    def __init__ (self):
-        """Initialize follow and index flags."""
-        super(MetaRobotsFinder, self).__init__()
-        log.debug(LOG_CHECK, "meta robots finder")
-        self.follow = self.index = True
-
-    def start_element (self, tag, attrs, element_text, lineno, column):
-        """Search for meta robots.txt "nofollow" and "noindex" flags."""
-        if tag == 'meta' and attrs.get('name') == 'robots':
-            val = attrs.get('content', u'').lower().split(u',')
-            self.follow = u'nofollow' not in val
```

Comment:
I want to remark that the old code did handle substring matches a la `<meta name="robots" content="...,nofollow,...">`, but it could be tricked by adding spaces after commas!

According to https://moz.com/learn/seo/robots-meta-directives, spaces after commas are allowed.

The `.lower()` is surprising to me, I would't have realized this directive is case-insensitive. More unit tests needed to cover all the cases!

Reply:
As above.